test/acceptance/tests/connect/health_checks_test.go

```diff
@@ -0,0 +1,69 @@
+package connect
+
+import (
+	"fmt"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"strconv"
+	"testing"
+)
+
+// Test that health checks work in a default installation and a secure installation with TLS/auto-encrypt permutations.
+// Deploy with a passing health check
+// Test that traffic passes
+// update the container with readiness probe so that it fails
+// Test that traffic now fails
```

Comment:
```suggestion
// Deploy with a passing health check.
// Test that the service is accessible over the mesh.
// Update the container with readiness probe so that it fails.
// Test that the service is inaccessible over the mesh.
```